### ISSUE TITLE
Remove `libavif` dependency

### ISSUE THREAD

**Issue by Aloso** [ArturKovacs/emulsion#183]
Currently emulsion uses a statically linked C library for AVIF support. Because this makes the build process more complicated and dependant on the build platform, it is feature gated.

Recently the image crate added AVIF decoding support, so it's now possible to remove the C dependency and get AVIF support on all platforms while simplifying the build process.

The downside is that the image crate currently only supports 8-bit color depth. AVIF files with 10-bit or 12-bit color depth can't be decoded. I'm not sure how big that problem is, it depends on how commonly used such files are in practice.

**Comment by ArturKovacs** [comment]
Considering that I have not encountered an AVIF file in the wild and you were the one who made the original PR for adding avif support, I will defer to your decision on this one. I do like the idea of removing C and Assemby dependencies.

**Comment by linkmauve** [comment]
Note that the image crate depends on [dav1d](https://crates.io/crates/dav1d) to do the decoding, which is also a C library.  The reason 10- and 12-bit files are unsupported for now is because I’m using [dcv-color-primitives](https://crates.io/crates/dcv-color-primitives) for the YUV→RGB conversion.  [An issue is open](https://github.com/aws/dcv-color-primitives/issues/42) for this feature.

**Comment by Aloso** [comment]
I think a C library is okay, as long as it is built automatically and doesn't require the library to be installed on the build machine. That makes the installation much easier.

**Comment by ArturKovacs** [comment]
I agree. It seems that the dav1d bindings, by default, expect the native library to be installed on the system, but seemingly this can be changed so that dav1d is statically linked which we would want at least on Windows and macOS. (`SYSTEM_DEPS_DAV1D_BUILD_INTERNAL=always`)

Maybe we should wait until 10 and 12 bit AVIF support gets added to the `image` crate.
